### ISSUE TITLE
declaration-empty-line-before doesn't ignores shared-line comments

### ISSUE THREAD

**Issue by agordeev-zz** [stylelint/stylelint#2847]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Now, declaration-empty-line-before  doesn't ignores shared-line comments if it's in a selector. Probably relates to #2237.

> Which rule, if any, is this issue related to?

`declaration-empty-line-before`
except: ["after-declaration", "first-nested"]
ignore: ["after-comment']

> What CSS is needed to reproduce this issue?

```css
.class { /* comment */
    display: block;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "declaration-empty-line-before": ["always", {
      except: [
          "after-declaration",
          "first-nested"
      ],
      ignore: [
          "after-comment"
      ]
  }]
}
```

> Which version of stylelint are you using?

`8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
48:9   ✖  Expected empty line before declaration   declaration-empty-line-before
```

**Comment by jeddy3** [comment]
@agordeev-zz Thanks for using the template. This should [be fixed](https://github.com/stylelint/stylelint/pull/2827) in the upcoming `8.1.0` release.
